tensorflow/lite/micro/integration_tests/generate_per_layer_tests.py

```diff
@@ -143,43 +141,106 @@ def __init__(self, output_dir, model_paths, inputs):
     self.model_paths = model_paths
     self.csv_filenames = []
     self.inputs = inputs
+    self.input_types = {}
     self.cc_srcs = []
     self.cc_hdrs = []
     self.includes = []
 
-  def generate_inputs_conv(self, interpreter):
+  def generate_inputs_single(self, interpreter, dtype):
     input_tensor = interpreter.tensor(
         interpreter.get_input_details()[0]['index'])
     return [
-        np.random.randint(low=-128,
-                          high=127,
-                          dtype=np.int8,
-                          size=input_tensor().shape)
+        np.random.randint(low=np.iinfo(dtype).min,
+                          high=np.iinfo(dtype).max,
+                          dtype=dtype,
+                          size=input_tensor().shape),
     ]
 
+  def generate_inputs_add_sub(self, interpreter, dtype):
+    input_tensor0 = interpreter.tensor(
+        interpreter.get_input_details()[0]['index'])
+    input_tensor1 = interpreter.tensor(
+        interpreter.get_input_details()[1]['index'])
+    return [
+        np.random.randint(low=np.iinfo(dtype).min,
+                          high=np.iinfo(dtype).max,
+                          dtype=dtype,
+                          size=input_tensor0().shape),
+        np.random.randint(low=np.iinfo(dtype).min,
+                          high=np.iinfo(dtype).max,
+                          dtype=dtype,
+                          size=input_tensor1().shape)
+    ]
+
+  def generate_inputs_transpose_conv(self, interpreter, dtype):
```

Comment:
All these generate_inputs_xxx function seems to be generating input tensors according to each OP's specific interface.  I imagine these will keep on growing and wonder whether this would be easy to maintain. WDYT of refactoring this to be a single GenerateInputForOp class in a sperate file and takes op and interpreter and so on?

Reply:
My hope is that we can mostly share implementations between ops. For example, all single-input single-output ops can share one implementation, most 2-intput 1 output ops share (add, sub, mul, div, etc) and only a few special ops need a specific impl (transpose_conv).

I'll move this into a separate class / file like you suggested.